builtin/mainmenu/tab_online.lua

```diff
@@ -264,19 +265,52 @@ end
 
 --------------------------------------------------------------------------------
 
+-- Separates by space characters and handles special prefixes
+-- (words with special prefixes need an exact match and none of them can contain spaces
+-- or quotation marks)
+local function parse_search_word(query, text)
+	for word in text:gmatch("%S+") do
+		if string.sub(word, 0, 4) == "mod:" then
+			table.insert(query.mods, string.sub(word, 5))
+		elseif string.sub(word, 0, 7) == "player:" then
+			table.insert(query.players, string.sub(word, 8))
+		elseif string.sub(word, 0, 5) == "game:" then
+			query.game = string.sub(word, 6)
+		else
+			table.insert(query.keywords, word:lower())
+		end
+	end
+end
+
+local function parse_search_input(input)
+	local query = {keywords = {}, mods = {}, players = {}}
+
+	-- Separate quote enclosed parts
+	-- (If the input contains a single quotation mark, the part after is ignored.)
+	parse_search_word(query, string.match(input, "^([^\"]*)"))
+	for s1, s2 in input:gmatch("\"([^\"]*)\"([^\"]*)") do
+		if #s1 > 0 then
+			table.insert(query.keywords, s1:lower()) -- quotation enclosed part keeps spaces
+		end
+		parse_search_word(query, s2)
+	end
+
+	if #query.keywords == 0 and #query.mods == 0 and #query.players == 0 and not query.game then
```

Comment:
Isn't this condition equivalent to `input:find("%S")` (non-space character in input)?

Reply:
Yes, I chanhed it.
(This woulnt be the case if there would be a emtiness check for the prefixes.)